### PR TITLE
ISPN-7635 Make log4j a test-scoped dependency of infinispan-core

### PR DESCRIPTION
**Pull request by tristantarrant**
- Also fixes a hang when generating component metadata under IBM JDK

https://issues.jboss.org/browse/ISPN-7635

### CODE REVIEW

**Comment by Sanne** [comment]
shouldn't this be two separate issues?

**Comment by danberindei** [comment]
@Sanne AFAICT the metadata persister changes are cosmetic only, but I agree it should be at least a separate commit.

File: parent/pom.xml [hunk 1]
```diff
@@ -1458,6 +1458,10 @@
                            <key>infinispan.isCoreModule</key>
                            <value>${module.metadata.isCoreModule}</value>
                         </systemProperty>
+                        <systemProperty>
+                           <key>log4j.configurationFile</key>
+                           <value>${log4j.configurationFile}</value>
+                        </systemProperty>
```
Comment (danberindei) [inline parent/pom.xml]: I'm not sure about this, as running the tool could overwrite the test suite's log file... True, the tool should run before the tests, and `infinispan.module-suffix` is not defined here, but I'd prefer having a tools-only log4j2 configuration that just writes to the console.

Comment (tristantarrant) [inline parent/pom.xml]: That is actually not needed anymore, since I've removed log4j from the classpath

**Comment by tristantarrant** [comment]
@Sanne removing log4j from the "compile" classpath of core (where it isn't needed) is what actually fixes the hang. As @danberindei says, the other changes are merely cosmetic, but I'll split them up.

**Comment by tristantarrant** [comment]
Fixed

**Comment by Sanne** [comment]
> Fixed

Thanks :) someone need to keep an eye on this project lead!

**Comment by danberindei** [comment]
The OSGi test suite hanged, I triggered another build.

**Comment by danberindei** [comment]
@tristantarrant The latest build also hanged:
http://ci.infinispan.org/viewLog.html?buildId=52851&tab=buildResultsDiv&buildTypeId=bt9

**Comment by danberindei** [comment]
@tristantarrant The initial error is:

```
2017-03-20 15:30:00,926 | ERROR | FelixStartLevel  | BootFeaturesInstaller            | 20 - org.apache.karaf.features.core - 3.0.4 | Error installing boot features
java.lang.Exception: Could not start bundle mvn:org.infinispan/infinispan-embedded/9.0.0-SNAPSHOT in feature(s) infinispan-embedded-9.0.0-SNAPSHOT: Unresolved constraint in bundle org.infinispan.embedded [46]: Unable to resolve 46.0: missing requirement [46.0] osgi.wiring.package; (osgi.wiring.package=org.apache.logging.log4j.core)
```

and the dependency seems to come from Hibernate, based on https://github.com/infinispan/infinispan/blob/master/all/embedded/src/main/resources/features.xml#L25

**Comment by tristantarrant** [comment]
I have cleaned up the OSGi features files as well as making some tweaks to the integration tests. Let's wait for CI.

**Comment by tristantarrant** [comment]
Looks good now

**Comment by danberindei** [comment]
Integrated, thanks Tristan!

**Pull request closed without merging**